### ISSUE TITLE
Add support for flow-switch parameter

### ISSUE THREAD

**Issue by kopierschnitte** [MiRo1310/ioBroker.midas-aquatemp#16]
In order to monitor the water flow through the heatpump, it might be handy to query the "Flow-Switch" measurement S03/S3.
States are CLOSE (water flowing through the heatpump) or OPEN (no water flow).

**Comment by MiRo1310** [comment]
Moin. Ich werde mir das mal anschauen.

**Comment by kopierschnitte** [comment]
Super ... wollte es erst selber probieren, aber ich will es erstmal noch testen. Finde die Angabe recht sinnvoll, um abzuschätzen, ob die Wasserpumpe funktioniert und läuft ;-)

**Comment by MiRo1310** [comment]
👍

**Comment by MiRo1310** [comment]
@kopierschnitte Ich habe das gerade mal integriert, aber leider scheint es so zu sein das der Wert nur überwacht wird wenn die Wärmepumpe an ist. Muss ich ab 10 Uhr mal schauen ob der Wert dann auf true geht

**Comment by MiRo1310** [comment]
@kopierschnitte Bei mir wird es nicht mal in der App gesetzt. Funktioniert das denn bei dir?

**Comment by kopierschnitte** [comment]
Cool. danke. Komischerweise bleiben beim aktuellen Code jedoch alle Datenpunkte bis auf TempSet bei mir 0...
Login und Auslesen der Daten scheint aber okay:


midas-aquatemp.0 | 2024-08-06 06:55:50.836 | info | DeviceDetails: {"sessionid":null,"error_code":"0","error_msg":"Success","error_msg_code":"","totalSize":null,"totalPage":null,"nextPage":null,"objectResult":[{"value":"1","code":"Power"},{"value":"1","code":"Mode"},{"value":"0","code":"Manual-mute"},{"value":"15.0","code":"T01"},{"value":"15.5","code":"T02"},{"value":"0000001000000000","code":"2074"},{"value":"0000000000001000","code":"2075"},{"value":"0000000000000000","code":"2076"},{"value":"0000000000000000","code":"2077"},{"value":"0","code":"H03"},{"value":"32.0","code":"Set_Temp"},{"value":"8.0","code":"R08"},{"value":"35.0","code":"R09"},{"value":"15.0","code":"R10"},{"value":"35.0","code":"R11"},{"value":"32.0","code":"R01"},{"value":"32.0","code":"R02"},{"value":"32.0","code":"R03"},{"value":"18.0","code":"T03"},{"value":"0","code":"1158"},{"value":"0","code":"1159"},{"value":"0","code":"F17"},{"value":"1","code":"H02"},{"value":"13.5","code":"T04"},{"value":"13.0","code":"T05"},{"value":"15.5","code":"T06"},{"value":"0.2","code":"T07"},{"value":"236","code":"T14"},{"value":"0","code":"T17"},{"value":"0","code":"T1"},{"value":"0","code":"T2"},{"value":"0","code":"T3"},{"value":"0","code":"T4"},{"value":"0","code":"T5"},{"value":"0","code":"T6"},{"value":"0","code":"T7"}],"isReusltSuc":true}
-- | -- | -- | --
midas-aquatemp.0 | 2024-08-06 06:54:50.613 | info | DeviceDetails: {"sessionid":null,"error_code":"0","error_msg":"Success","error_msg_code":"","totalSize":null,"totalPage":null,"nextPage":null,"objectResult":[{"value":"1","code":"Power"},{"value":"1","code":"Mode"},{"value":"0","code":"Manual-mute"},{"value":"15.0","code":"T01"},{"value":"15.5","code":"T02"},{"value":"0000001000000000","code":"2074"},{"value":"0000000000001000","code":"2075"},{"value":"0000000000000000","code":"2076"},{"value":"0000000000000000","code":"2077"},{"value":"0","code":"H03"},{"value":"32.0","code":"Set_Temp"},{"value":"8.0","code":"R08"},{"value":"35.0","code":"R09"},{"value":"15.0","code":"R10"},{"value":"35.0","code":"R11"},{"value":"32.0","code":"R01"},{"value":"32.0","code":"R02"},{"value":"32.0","code":"R03"},{"value":"18.0","code":"T03"},{"value":"0","code":"1158"},{"value":"0","code":"1159"},{"value":"0","code":"F17"},{"value":"1","code":"H02"},{"value":"13.5","code":"T04"},{"value":"13.0","code":"T05"},{"value":"15.5","code":"T06"},{"value":"0.2","code":"T07"},{"value":"236","code":"T14"},{"value":"0","code":"T17"},{"value":"0","code":"T1"},{"value":"0","code":"T2"},{"value":"0","code":"T3"},{"value":"0","code":"T4"},{"value":"0","code":"T5"},{"value":"0","code":"T6"},{"value":"0","code":"T7"}],"isReusltSuc":true}
midas-aquatemp.0 | 2024-08-06 06:54:50.266 | info | Login ok! Token: qdrAguBxnCCuuLVTY7i6mT2/r1h85F9sEdGYuU/8E/TegIE+qIkhL8O2zY67tLpT+PwWYJ9m18VpYiaaf56BlA==



Es scheint jedoch zu klappen, wenn ich das Auslesen nicht mithilfe der MAC-Adresse starte. Ohne MAC kommen die Werte an und das Log zeigt folgendes:


midas-aquatemp.0 | 2024-08-06 07:00:38.636 | info | DeviceDetails: {"sessionid":null,"error_code":"0","error_msg":"Success","error_msg_code":"","totalSize":null,"totalPage":null,"nextPage":null,"objectResult":[{"value":"1","code":"Power"},{"value":"1","code":"Mode"},{"value":"0","code":"Manual-mute"},{"value":"15.0","code":"T01"},{"value":"15.5","code":"T02"},{"value":"0000001000000000","code":"2074"},{"value":"0000000000001000","code":"2075"},{"value":"0000000000000000","code":"2076"},{"value":"0000000000000000","code":"2077"},{"value":"0","code":"H03"},{"value":"32.0","code":"Set_Temp"},{"value":"8.0","code":"R08"},{"value":"35.0","code":"R09"},{"value":"15.0","code":"R10"},{"value":"35.0","code":"R11"},{"value":"32.0","code":"R01"},{"value":"32.0","code":"R02"},{"value":"32.0","code":"R03"},{"value":"18.0","code":"T03"},{"value":"0","code":"1158"},{"value":"0","code":"1159"},{"value":"0","code":"F17"},{"value":"1","code":"H02"},{"value":"13.5","code":"T04"},{"value":"13.0","code":"T05"},{"value":"15.5","code":"T06"},{"value":"0.2","code":"T07"},{"value":"236","code":"T14"},{"value":"0","code":"T17"},{"value":"0","code":"T1"},{"value":"0","code":"T2"},{"value":"0","code":"T3"},{"value":"0","code":"T4"},{"value":"0","code":"T5"},{"value":"0","code":"T6"},{"value":"0","code":"T7"}],"isReusltSuc":true}
-- | -- | -- | --

**Comment by MiRo1310** [comment]
Ich habe das mit Flow Switch noch garnicht hoch geladen. Daher kann es auch nicht funktionieren. Mache ich aber gleich.

**Comment by kopierschnitte** [comment]
Keine Eile ;-)

**Comment by MiRo1310** [comment]
Was meintest du genau mit Werte bleiben 0? Nur die Datenpunkte bleiben 0 ? Weil im Json ja überall z.b. T05 und T04 drin sind. Ich habe das mit der Mac Adresse selbst auch noch garnicht an meiner Wärmepumpe getestet , nur an der von ...

**Comment by kopierschnitte** [comment]
Genau, die DP blieben dann auf 0. Bei der "klassischen" Anmeldung mit selbständiger Abfrage der DeviceID war alles ok.

**Comment by MiRo1310** [comment]
@kopierschnitte  Ich habe jetzt die Funktion findCodeVal umgebaut. Du kannst jetzt ein Array mit Werten übergeben , z.b. ["T1", "T01"] wobei die Funktion dann schaut welcher Eintrag einen Wert hat der nicht 0 ist und diesen zurück gibt, wenn kein Wert !== "0" ist wird einfach "0" zurück gegeben. Somit brauchen wir nicht mehr überprüfen ob es sich um eine PoolsanaId handelt. Somit sollte das mit der MacId jetzt immer funktionieren

**Comment by kopierschnitte** [comment]
Geniale Idee. Vor dem Problem stand ich auch immer. Klingt nach einer wirklich kreativen Lösung!

**Comment by MiRo1310** [comment]
Ist alles hoch geladen, du kannst testen. Ich werde schon mal ein neues Release machen. Wenn dann alles funktioniert kümmere ich mich darum das wird den Adapter schon mal ins latest rein bekommen
